File: shared/coordinates.py
from dataclasses import dataclass


@dataclass
class HitBox:
    x_start: int
    x_end: int
    y_start: int
    y_end: int


@dataclass
class TouchCoordinates:
    x: int
    y: int
    s: int

    def check_hit(self, hbox: HitBox) -> bool:
        if self.x < hbox.x_start or self.x > hbox.x_end:
            return False
        if self.y < hbox.y_start or self.y > hbox.y_end:
            return False

        return True

File: v2/shared/coordinates.go
package shared

type HitBox struct {
	XStart int
	XEnd   int
	YStart int
	YEnd   int
}

type TouchCoordinates struct {
	X int
	Y int
	S int
}

func NewTouchCoordinates(x, y, s int) TouchCoordinates {
	return TouchCoordinates{
		X: x,
		Y: y,
		S: s,
	}
}

func (c TouchCoordinates) In(hb HitBox) bool {
	if c.X < hb.XStart || c.X > hb.XEnd {
		return false
	}
	if c.Y < hb.YStart || c.Y > hb.YEnd {
		return false
	}
	return true
}
